headers/types/common/enums.h

```diff
@@ -3726,6 +3726,362 @@ enum mission_rank {
+    MISSION_REWARD_ITEM = 2,
+    MISSION_REWARD_ITEM_AND_MORE = 3, // Item + (?)
+    // Exclusive Item, displayed as "(?)" (Will be 1 star, and beneficial of the client/target)
+    // $SCENARIO_MAIN must be [8, 0] or above AND mission rank must be >=5 stars to be selected.
+    // Also used for the Togetic Wing given from the Gabite Scale mission.
+    MISSION_REWARD_EXCLUSIVE_ITEM = 4,
+    // Money, displayed as "(?)"
+    // $SCENARIO_MAIN must be [12, 0] or above to be selected.
+    MISSION_REWARD_MONEY_HIDDEN = 5,
+    // Either an egg or the client requests to join the team, displayed as "(?)"
+    // $SCENARIO_MAIN must be [6, 0] or above for this to be selected
+    MISSION_REWARD_SPECIAL = 6,
+};
+// This is usually stored as an 8-bit integer
+#pragma pack(push, 1)
+ENUM_8_BIT(mission_reward_type);
+#pragma pack(pop)
+
+// Different types of restrictions that a mission can have
+enum mission_restriction_type {
+    MISSION_RESTRICTION_NONE = 0,
+    MISSION_RESTRICTION_TYPE = 1,    // Requires a pokémon of a certain type on the team
+    MISSION_RESTRICTION_MONSTER = 2, // Requires a certain pokémon on the team
+};
+// This is usually stored as an 8-bit integer
+#pragma pack(push, 1)
+ENUM_8_BIT(mission_restriction_type);
+#pragma pack(pop)
+
+// The restriction of a mission can be a monster ID or a type ID
+union mission_restriction {
+    struct monster_id_16 monster_id;
+    struct type_id_8 type_id;
+};
+
+// Represents the different statuses that a mission can have
+enum mission_status {
+    MISSION_STATUS_INVALID = 0, // Used for empty mission slots
+    MISSION_STATUS_UNK_1 = 1,   // The mission won't display a status
+    MISSION_STATUS_UNK_2 = 2,   // The mission won't display a status
+    MISSION_STATUS_UNK_3 = 3,   // The mission won't display a status
+    MISSION_STATUS_SUSPENDED = 4,
+    MISSION_STATUS_ACCEPTED = 5,
+    MISSION_STATUS_DONE = 6,
+    MISSION_STATUS_UNK_7 = 7, // Shows up as "accepted"
+    MISSION_STATUS_UNK_8 = 8, // Shows up as "accepted"
+};
+// This is usually stored as an 8-bit integer
+#pragma pack(push, 1)
+ENUM_8_BIT(mission_status);
+#pragma pack(pop)
+
+enum mission_template_item_case {
+    // Treated same as ITEM_CASE_SPECIFIC_ITEM, unused in template
+    ITEM_CASE_UNK_0 = 0,
+    // mission_template_item_data_1 is ignored, and mission_template_item_data_2 is an item_id
+    ITEM_CASE_SPECIFIC_ITEM = 1,
+    // mission_template_item_data_1 and mission_template_item_data_2 are indices to a table
+    ITEM_CASE_USE_ITEM_TABLE = 2,
+    // Has a distinct case menu, unused in template.
+    ITEM_CASE_UNK_3 = 3,
+    // mission_template_item_data_1 and mission_template_item_data_2 are ignored
+    ITEM_CASE_NO_ITEM = 4
+};
+
+#pragma pack(push, 2)
+ENUM_16_BIT(mission_template_item_case);
+#pragma pack(pop)
+
+union mission_template_item_data_1 {
+    uint16_t none;
+    uint16_t template_item_table_id; // if ITEM_CASE_USE_ITEM_TABLE
+};
+
+union mission_template_item_data_2 {
```

Comment:
This "1" and "2" business seems excessive to me. They're essentially the same union, just that some fields will never take certain variant values. I doubt the game has separate code to deal with 1 vs. 2. I would rather we just define one permissive union type here and then note the in-game restrictions on the values in the struct field comments. That's simpler and more comprehensible IMO.